lib/modules/manager/gitlabci/extract.ts

```diff
@@ -2,96 +2,95 @@ import is from '@sindresorhus/is';
 import { load } from 'js-yaml';
 import { logger } from '../../../logger';
 import { readLocalFile } from '../../../util/fs';
-import { newlineRegex, regEx } from '../../../util/regex';
+import { regEx } from '../../../util/regex';
 import { getDep } from '../dockerfile/extract';
 import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
-import type { GitlabPipeline } from './types';
+import type { GitlabPipeline, Image, JobTemplate, Services } from './types';
 import { replaceReferenceTags } from './utils';
 
-const commentsRe = regEx(/^\s*#/);
-const aliasesRe = regEx(`^\\s*-?\\s*alias:`);
-const whitespaceRe = regEx(`^(?<whitespace>\\s*)`);
-const imageRe = regEx(
-  `^(?<whitespace>\\s*)image:(?:\\s+['"]?(?<image>[^\\s'"]+)['"]?)?\\s*$`
-);
-const nameRe = regEx(`^\\s*name:\\s+['"]?(?<depName>[^\\s'"]+)['"]?\\s*$`);
-const serviceRe = regEx(
-  `^\\s*-?\\s*(?:name:\\s+)?['"]?(?<depName>[^\\s'"]+[^:]$)['"]?\\s*$`
-);
-function skipCommentAndAliasLines(
-  lines: string[],
-  lineNumber: number
-): { lineNumber: number; line: string } {
-  let ln = lineNumber;
-  while (
-    ln < lines.length - 1 &&
-    (commentsRe.test(lines[ln]) || aliasesRe.test(lines[ln]))
-  ) {
-    ln += 1;
+export function extractFromImage(image: Image | undefined): PackageDependency {
+  if (is.undefined(image)) {
+    return undefined;
   }
-  return { line: lines[ln], lineNumber: ln };
+  let dep: PackageDependency = {};
+  if (is.string(image)) {
+    dep = getDep(image);
+    dep.depType = 'image';
+    return dep;
+  }
+  dep = getDep(image.name);
+  dep.depType = 'image-name';
+  return dep;
 }
 
-export function extractPackageFile(content: string): PackageFile | null {
+export function extractFromServices(
+  services: Services | undefined
+): PackageDependency[] {
+  if (is.undefined(services)) {
+    return undefined;
+  }
+
   const deps: PackageDependency[] = [];
+  let dep: PackageDependency<Record<string, any>> = {};
+  for (const s of services) {
```

Comment:
```suggestion
  for (const service of services) {
```

Better to use longer variable names where possible